File: serve.js
/* 极简静态服务器（无依赖），供 start-server.bat 在找不到 python 时兜底 */
const http = require('http');
const fs = require('fs');
const path = require('path');

const ROOT = path.join(__dirname, 'src');
const PORT = Number(process.env.PORT || 8080);
const MIME = {
  '.html':'text/html; charset=utf-8', '.htm':'text/html; charset=utf-8',
  '.js':'text/javascript; charset=utf-8', '.css':'text/css; charset=utf-8',
  '.json':'application/json; charset=utf-8',
  '.webp':'image/webp', '.png':'image/png', '.jpg':'image/jpeg', '.jpeg':'image/jpeg',
  '.gif':'image/gif', '.svg':'image/svg+xml', '.ico':'image/x-icon',
  '.woff2':'font/woff2', '.mp3':'audio/mpeg', '.zip':'application/zip'
};

http.createServer((req, res) => {
  let p = decodeURIComponent(req.url.split('?')[0]);
  if (p === '/') p = '/index.html';
  const file = path.join(ROOT, path.normalize(p).replace(/^([/\\])+/, ''));
  if (!file.startsWith(ROOT)) { res.writeHead(403); return res.end('forbidden'); }
  fs.readFile(file, (err, buf) => {
    if (err) { res.writeHead(404, {'Content-Type':'text/plain; charset=utf-8'}); return res.end('404 ' + p); }
    res.writeHead(200, { 'Content-Type': MIME[path.extname(file).toLowerCase()] || 'application/octet-stream', 'Cache-Control':'no-cache' });
    res.end(buf);
  });
}).listen(PORT, () => {
  console.log('拾光·澈屿 → http://localhost:' + PORT + '/index.html');
});

File: serve.py
"""极简静态服务器：把 src/ 目录挂到 http://localhost:8080
用法：python serve.py   （或双击 start-server.bat）"""
import http.server
import os
import socketserver

PORT = int(os.environ.get("PORT", 8080))
ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "src")


class Handler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=ROOT, **kwargs)

    def end_headers(self):
        self.send_header("Cache-Control", "no-cache, no-store, must-revalidate")
        super().end_headers()

    def log_message(self, fmt, *args):
        pass


if __name__ == "__main__":
    socketserver.TCPServer.allow_reuse_address = True
    with socketserver.TCPServer(("", PORT), Handler) as httpd:
        print("Q版换装小游戏 -> http://localhost:%d/index.html" % PORT)
        print("关闭本窗口即停止服务。")
        httpd.serve_forever()
